### PR TITLE
fixing error with portals

### PR DESCRIPTION
**Pull request by antgonza**
An error was identified when looking at the portals list. The error is simple to solve but no idea how to test it as the error was due to the changes we made (now returning objects vs. ids). BTW I think we had a discussion about the get_info method before and agreed that it was fine to just work with ids.

### CODE REVIEW

**Comment by squirrelo** [comment]
You can test by using mock to log in as admin@foo.bar, then doing the standard code and body tests like [here](https://github.com/biocore/qiita/blob/master/qiita_pet/test/test_study_handlers.py#L211)

**Comment by antgonza** [comment]
Thanks. It turns out that portals had no tests in qiita_pet. Added all (AFAIK). Ready for review.

File: qiita_pet/test/test_portal.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+from qiita_db.user import User
+from qiita_pet.handlers.base_handlers import BaseHandler
+
+
+class TestPortal(TestHandlerBase):
+    def test_get(self):
+        BaseHandler.get_current_user = Mock(return_value=User("admin@foo.bar"))
+        response = self.get('/admin/portals/studies/')
+        self.assertEqual(response.code, 200)
```
Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: needs test of `response.body`

File: qiita_pet/test/test_portal.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    def test_post(self):
+        BaseHandler.get_current_user = Mock(return_value=User("admin@foo.bar"))
+        response = self.post('/admin/portals/studies/', {'portal': 'EMP',
+                                                         'selected': [1],
+                                                         'action': 'Add'})
+        self.assertEqual(response.code, 200)
+
+        response = self.post('/admin/portals/studies/', {'portal': 'EMP',
```
Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: Break this into two tests.

File: qiita_pet/test/test_portal.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        BaseHandler.get_current_user = Mock(return_value=User("admin@foo.bar"))
+        response = self.get('/admin/portals/studies/')
+        self.assertEqual(response.code, 200)
+
+    def test_post(self):
+        BaseHandler.get_current_user = Mock(return_value=User("admin@foo.bar"))
+        response = self.post('/admin/portals/studies/', {'portal': 'EMP',
+                                                         'selected': [1],
+                                                         'action': 'Add'})
+        self.assertEqual(response.code, 200)
```
Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: needs test of `response.body`

File: qiita_pet/test/test_portal.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+        response = self.post('/admin/portals/studies/', {'portal': 'EMP',
+                                                         'selected': [1],
+                                                         'action': 'Remove'})
+        self.assertEqual(response.code, 200)
```
Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: needs test of `response.body`

**Comment by squirrelo** [comment]
Some comments.

File: qiita_pet/test/test_portal.py (hunk 1 above)
Comment (antgonza) [inline qiita_pet/test/test_portal.py]: This applies to all response.body comments. What should I tests? That the display is accurate all html (too much IMOO, as any minimal change will make the test fail), that the response has the study id/title, other?

File: qiita_pet/test/test_portal.py (hunk 2 above)
Comment (antgonza) [inline qiita_pet/test/test_portal.py]: Out of curiosity, why do you think is needed? Note that these are really simple tests and seem like too much to have a test for each case, for example take a look at test_*_errors

Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: We agreed before that we were going to do atomic tests, with one action tested per test function.

File: qiita_pet/test/test_portal.py (hunk 1 above)
Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: Some line of the HTML that is dynamically generated from data is proper, like in the example I posted. We have had cases of 200 code with an empty html because of rendering issues.

Comment (antgonza) [inline qiita_pet/test/test_portal.py]: "Funny" enough there is nothing dynamic created in this page as it's retrieved via AJAX, and JS is not executed. Everything is created in /admin/portals/studiesAJAX/, for example: /admin/portals/studiesAJAX/?sEcho=1001&view-portal=QIITA ...

**Comment by josenavas** [comment]
@antgonza I think this is fixing https://github.com/biocore/qiita/issues/1616 correct?

**Comment by antgonza** [comment]
Yup, missed that one. Is there one about adding tests to portals? :yum:

File: qiita_pet/test/test_portal.py (hunk 1 above)
Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: The JSON it returns is dynamic. Please test that it comes out correctly.

Comment (josenavas) [inline qiita_pet/test/test_portal.py]: Agree with @squirrelo . The easiest way is to grab some static text from the template and check it in the response. In this case, it will suffice to do: `self.assertIn("View from portal:", response.body)`

File: qiita_pet/test/test_portal.py (hunk 3 above)
Comment (josenavas) [inline qiita_pet/test/test_portal.py]: Which you can use the same as above.

File: qiita_pet/test/test_portal.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+                                                         'selected': [1],
+                                                         'action': 'Add'})
+        self.assertEqual(response.code, 200)
+
+    def test_post_remove(self):
+        BaseHandler.get_current_user = Mock(return_value=User("admin@foo.bar"))
+        response = self.post('/admin/portals/studies/', {'portal': 'EMP',
+                                                         'selected': [1],
+                                                         'action': 'Remove'})
+        self.assertEqual(response.code, 200)
```
Comment (josenavas) [inline qiita_pet/test/test_portal.py]: And here.

**Comment by josenavas** [comment]
@antgonza I don't think so, my guess is that we englobed all missing tests in the interface in the general issue.

Just a small comment on the tests.

Thanks!

File: qiita_pet/test/test_portal.py (hunk 1 above)
Comment (antgonza) [inline qiita_pet/test/test_portal.py]: @squirrelo, you are right, the JSON is dynamic. However, as stated in my previous comment the AJAX call is not executed as it is in JS. Thus, the page never "renders" the study list == no way to test in here. The only option is to add tests to /admin/portals/studiesAJAX/, which is way beyond the issue this PR is solving but I could do it.

Comment (antgonza) [inline qiita_pet/test/test_portal.py]: Just realized that adding a test for the AJAX page is not that difficult. Can do it here.

Comment (antgonza) [inline qiita_pet/test/test_portal.py]: After chatting offline with @josenavas we agreed on adding a test for not empty body when reply is 200. Adding now but I want to discuss this in the next meeting (already in the list) because we should be doing something smarter than adding tests that test that there are specific words in the rendering.

**Comment by josenavas** [comment]
:+1: once tests pass

File: qiita_pet/test/test_portal.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        self.assertNotEqual(response.code, "")
+
+    def test_post_add(self):
+        BaseHandler.get_current_user = Mock(return_value=User("admin@foo.bar"))
+        response = self.post('/admin/portals/studies/', {'portal': 'EMP',
+                                                         'selected': [1],
+                                                         'action': 'Add'})
+        self.assertEqual(response.code, 200)
+        self.assertNotEqual(response.code, "")
```
Comment (squirrelo) [inline qiita_pet/test/test_portal.py]: Shouldn't this be `response.body`? Same for the other 2.

Comment (antgonza) [inline qiita_pet/test/test_portal.py]: Good catch!

**Comment by antgonza** [comment]
I think this is ready for merge.
